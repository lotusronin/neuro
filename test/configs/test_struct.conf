conf = {
'src' : 'test_struct.nro',
'expected' : 'test_struct.expected'
}
