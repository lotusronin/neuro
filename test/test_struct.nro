import stdio;

struct point {
    x : u32;
    y : u32;
    z : u32;
}

fn main() : u32 {
    foo : point;
    foo.x = 0;
    foo.y = 1;
    foo.z = 2;
    stdio::puts("point at (x,y,z)");
    stdio::printi(foo.x);
    stdio::putchar(',');
    stdio::printi(foo.y);
    stdio::putchar(',');
    stdio::printi(foo.z);
    stdio::putchar('\n');
    return 0;
}
