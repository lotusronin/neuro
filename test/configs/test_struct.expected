point at (x,y,z)
0,1,2
